miden-fmt/tests/expected/procedure_formatting_formatted.masm

```diff
@@ -0,0 +1,46 @@
+# Test procedure formatting: indentation, spacing, documentation
+
+use.miden::some_import
+
+#! A simple procedure that adds two numbers
+#!
+#! Stack: [a, b]
+#! Output: [a + b]
+export.add_two_numbers
+    push.1
+    push.2
+    add
+end
+
+#! A more complex procedure with nested operations
+#!
+#! Stack: [x, y, z]
+#! Output: [result]
+export.complex_procedure
+    dup.0
+    # Then perform some operations
+    push.10 mul
+    # => []
+
+    if.true
+        add.1
+        # => [val]
+    else
+        sub.1
+        # => [val]
+    end
+    add
+    # => [val]
```

Comment:
```suggestion
    if.true
        add.1
        # => [val]
    else
        sub.1
        # => [val]
    end

    add
    # => [val]
```
Perhaps it would make sense to insert newlines after `end`, _iff_ there is an instruction following?